### PR TITLE
Android: Add custom badge number

### PR DESCRIPTION
**Pull request by m1ga**
You can add `"badge": (int)` to a data notification and it will display that number instead of increment the badge count

https://github.com/hansemannn/titanium-firebase-cloud-messaging/issues/69

I'll test it over the weekend if I can get a phone with notification badges :-)
 
[Test file](http://migaweb.de/firebase.cloudmessaging-android-2.0.2.zip)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -174,6 +174,7 @@ Supported data fields:
 * "color" => will tint the app name and the small icon next to it
 * "vibrate" => "boolean"
 * "sound" => "string" (e.g. "notification.mp3" will play /platform/android/res/raw/notification.mp3)
+* "showBadge" => "int"
```
Comment (hansemannn) [inline README.md]: Change here as well. Also: If explicitly set to `null`, iOS resets the badge to 0. And not sure if a string-value `+1` or `-1` should also be accepted.

Comment (m1ga) [reply README.md]: ah, missed to push the updated README. I'll check if there is a difference between setting it to null and not setting it at all (then the system should take care of badges). Also I might need to use https://github.com/leolin310148/ShortcutBadger for all phones (e.g. OneSignal is using it too)

**Comment by hansemannn** [comment]
@miga How can we proceed here?

**Comment by m1ga** [comment]
I've pushed the ShortcutBadger part. It works but I'll want to do some more tests and need to add the `null` part to remove the badge

**Comment by m1ga** [comment]
did some tests with `null`. It already works: sending no `badge` parameter, `null` or `0` it will remove the number. Everything else will be shown as a value.

`+1` is not supported and judging by the sources you should manage incrementing numbers on the server. I *think* we could use
```
Activity activity = TiApplication.getAppCurrentActivity();
SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(activity);
```
to store the current value and implement a `+1` but it could be out of sync quickly since the icon number we be 0 if you just remove the notification and not open the app. A `+1` would show the old number + 1 and not 1.
So I think we should merge this one first and think about the `+1` a bit more

**Comment by m1ga** [comment]
#78 is almost ready too. So we can add this to `2.0.3` too
